pandas/core/generic.py

```diff
@@ -8564,6 +8523,179 @@ def nanptp(values, axis=0, skipna=True):
             ``numpy.ndarray`` method ``ptp``.""",
             nanptp)
 
+    _shared_docs['transform'] = ("""
+    Apply a function that returns a same-shaped object.
+
+    .. versionadded:: 0.20.0
+
+    Parameters
+    ----------
+    func : callable, list of callables, or dict
+        The function or function to apply to the DataFrame or Series.
+        Each function should expect a Series and return an array-like
+        with the same shape. Accepted combinations are:
+
+        - function : Applied to the Series, or to each column of the
+          DataFrame independently.
+        - list of functions : Each function is applied to the Series
+          or to each column of the DataFrame independently.
+        - dict of column names -> functions or list of functions :
+          For DataFrames only, apply the function or functions for a
+          given key to a specific column.
+
+    axis : {0 or 'index', 1 or 'columns'}, default 'index'
+        Axis to apply the function along (DataFrame only).
+
+        * 0 or 'index' : apply along the index, i.e. the function
+          gets each column.
+        * 1 or 'columns' : apply along the columns, i.e. the function
+          gets each row.
+
+    *args
+        Additional positional arguments are passed to `func`.
+    **kwargs
+        Additional keyword arguments are passed to `func`.
+
+    Returns
+    -------
+    transformed : Series or DataFrame
+        The result type is the same as the input. Furthermore, it is
+        the same shape.
+
+    Notes
+    ------
+    The function performs the transformation.
+    It takes a Series as input and must return a same length Series.
+    Note that the function may modify the Series index.
+
+    If applied to a Series the function is applied once.
+    If applied to a DataFrame the function is applied on each row or column.
+    Note that if the function modifies the index it must do so consistently
+    in the case of a DataFrame.
```

Comment:
I'd move that to the extended summary. And I would probably get rid of the first sentence.

Reply:
Removed first sentence.  
I don't know what/where the extended summary is... Notes seemed to me the most complementary section available. I could just delete the Notes section ?